### ISSUE TITLE
Use tycho pomless?

### ISSUE THREAD

**Issue by laeubi** [eclipse-equinox/equinox.bundles#37]
I noticed most bundles use trivial pom files so it might be a goof idea to use the pomless feature of tycho to e.g. prevent the need to adjust the version in two different files.

**Comment by HannesWell** [comment]
Yes definitely, I also had the intention to enable Tycho-'pomless' for this repo.
Many non-trivial poms only suppress compiler warnings and would become trivial if those warnings were fixed. Therefore I first had the intention to fix them, but probably its better if we (you or me?) start with removing those that already trivial and gradually fix away the need for the remaining non-trivial poms later.

**Comment by laeubi** [comment]
Will warnings fail the build? If no we could just remove this (only gives false impression of "everything is good), if yes, one could keep them and remove it part by part ...

**Comment by HannesWell** [comment]
> Will warnings fail the build? If no we could just remove this (only gives false impression of "everything is good), if yes, one could keep them and remove it part by part ...

I don't think so. They will just appear Build-logs (e.g. https://download.eclipse.org/eclipse/downloads/drops4/I20220501-1800/testResults.php) in the `Plugins containing compile errors or warnings` and `Plugins containing access errors or warnings`.
Some contributors scan them regularly and will inform us about new entries. But I agree that just suppressing them gives a false impressions, so I would be fine to fix them subsequently. I had the intention already because they also show up in the IDE and are relatively easy to fix (although there are many).

**Comment by laeubi** [comment]
> although there are many

If we remove the suppression and simply fix them whenever a file is changed I think that's suitable to make progress here...

**Comment by akurtakov** [comment]
We tried to keep this list empty (aka no new warnings appear) but it's fine to have temporary increase (aka enabling more warnings that were ignored before that) as long as there is dedicated effort to get to clean state ASAP. Reasoning behind that is if a list is not empty no one pays attention and keeps growing.
So I would recommend going incrementally here unless someone is ready to spend as much time as needed to get back to clean state.

**Comment by vogella** [comment]
Hidding the warnings via the build files does not help with preventing new warnings. The better approach would be to capture the current number of warnings somehow in the build, e.g. in the Jenkins file and fail the build if a new change ends up with more warnings. IIRC we had something like this in the PDE build do not see a releted configuration in the Jenkins file. Maybe this was lost during the migration.

**Comment by HannesWell** [comment]
> Hidding the warnings via the build files does not help with preventing new warnings. The better approach would be to capture the current number of warnings somehow in the build, e.g. in the Jenkins file and fail the build if a new change ends up with more warnings. IIRC we had something like this in the PDE build do not see a releted configuration in the Jenkins file. Maybe this was lost during the migration.

This would be definitely a good way. But there is a lot to fix. My IDE shows me a little bit more than 1000 warnings only in equinox bundles. 

However in the meantime I suggest to start to apply Tycho-build/pomless to those projects that already have a trivial pom. We can then gradually remove the other poms as it is suitable.

**Comment by vogella** [comment]
> However in the meantime I suggest to start to apply Tycho-build/pomless to those projects that already have a trivial pom
+1 that is what we also did in platform projects. 

If we would get a test / Jenkins verification to present the warnings to grow, my guess would be that @akurtakov would also be ok to remove the rest of the pom files.
